### PR TITLE
Feature/justiceotuya/adding redux

### PR DESCRIPTION
**Pull request by justiceotuya**

### CODE REVIEW

**Review by justiceotuya — COMMENTED** [review]
**I Am sorry the changes reached up to 2000, its just a file name change that caused this. i had to change the file names for it to pass travis**

File: components/layout/components/PageLayout.test.jsx [hunk 1]
```diff
@@ -3,7 +3,7 @@ import ReactDOM from 'react-dom';
 import NavHeader from './NavHeader';
 import PageFooter from './PageFooter';
 import Sidebar from './Sidebar';
-import PageLayout from './index';
+import PageLayout from './PageLayout';
```
Comment (olajohn-ajiboye) [inline components/layout/components/PageLayout.test.jsx:6]: Could you just make the import in alpha-numeric order according to the guideline?

import React from 'react';
import ReactDOM from 'react-dom';
import NavHeader from './NavHeader';
import PageFooter from './PageFooter';
import PageLayout from './PageLayout';
import Sidebar from './Sidebar';

**Review by olajohn-ajiboye — DISMISSED** [review]
Looks ok other than the requested change. Nice job overall.

File: lib/middleware/with-redux-store.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
+/* eslint-disable no-param-reassign */
+/* eslint-disable react/jsx-filename-extension */
```
Comment (Way2nnadi) [inline lib/middleware/with-redux-store.js:2]: remove `eslint-disable react/jsx-filename-extension`

File: components/timeLine/index.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
+/* eslint-disable import/prefer-default-export */
```
Comment (Way2nnadi) [inline components/timeLine/index.js]: remove this here and disable it in the `.eslintrc`

File: components/timeLine/components/TimeLinePopularTopic.jsx [hunk 4]
```diff
@@ -13,6 +13,7 @@ const TimeLinePopularTopic = () => (
                     return (
                         <li key={text}>
                             <Link href={link}>
+                                {/* eslint-disable-next-line jsx-a11y/anchor-is-valid */}
```
Comment (Way2nnadi) [inline components/timeLine/components/TimeLinePopularTopic.jsx]: remove this disable and only user the anchor tag is you actual have an `href` connected to it. don't abuse it for styling. just all a class to the `Link` and add the appropriate css

File: components/timeLine/components/TimeLineOnlineFriends.jsx [hunk 5]
```diff
@@ -16,9 +17,9 @@ const TimeLineOnlineFriends = () => (
                 return (
                     <li key={email}>
                         <img
-                          src={photo}
-                          alt="user's face"
-                          className="user-avatar avartar-online"
+                            src={photo}
+                            alt="user's face"
```
Comment (Way2nnadi) [inline components/timeLine/components/TimeLineOnlineFriends.jsx]: alt value should be a variable

File: components/timeLine/components/TimeLine.jsx [hunk 6]
```diff
@@ -170,8 +184,8 @@ class TimeLine extends React.Component {
                         <section style={{ background: 'white' }}>
```
Comment (Way2nnadi) [inline components/timeLine/components/TimeLine.jsx]: remove this in-line style

File: components/layout/components/PageFooter.jsx [hunk 7]
```diff
@@ -19,14 +20,18 @@ export default function PageFooter() {
                 <Footer className="LandingPage_footer">
                     <Link href="/">
                         <a>
-                            <img src="../../../static/logo-light.png" alt="helpme logo" className="logo" />
+                            <img
+                                src="../../../static/logo-light.png"
+                                alt="helpme logo"
```
Comment (Way2nnadi) [inline components/layout/components/PageFooter.jsx]: alt value as variable

File: components/layout/components/PageFooter.jsx [hunk 8]
```diff
@@ -19,14 +20,18 @@ export default function PageFooter() {
                 <Footer className="LandingPage_footer">
                     <Link href="/">
                         <a>
-                            <img src="../../../static/logo-light.png" alt="helpme logo" className="logo" />
+                            <img
+                                src="../../../static/logo-light.png"
```
Comment (Way2nnadi) [inline components/layout/components/PageFooter.jsx]: src value as variable

File: components/layout/components/NavHeader.jsx [hunk 9]
```diff
@@ -38,23 +42,23 @@ function NavHeader(props) {
                 <link rel="shortcut icon" href="../../static/favicon.ico" type="image/x-icon" />
                 <link rel="icon" href="../../static/favicon.ico" type="image/x-icon" />
                 <link
-                  rel="stylesheet"
-                  href="https://cdnjs.cloudflare.com/ajax/libs/normalize/8.0.1/normalize.min.css"
+                    rel="stylesheet"
+                    href="https://cdnjs.cloudflare.com/ajax/libs/normalize/8.0.1/normalize.min.css"
```
Comment (Way2nnadi) [inline components/layout/components/NavHeader.jsx]: save all the `href`'s in this file as variables in `constants.js`

File: components/layout/components/NavHeader.jsx [hunk 10]
```diff
@@ -38,23 +42,23 @@ function NavHeader(props) {
...
                 />
                 <title>{!title ? HEADER_TITLE : title}</title>
             </Head>
             {/* navheader for mobile */}
             <Header theme="light" className="layout_header-mobile">
-                <Link href="/">
+                <Link href="http://localhost:3000/api/users/login">
```
Comment (Way2nnadi) [inline components/layout/components/NavHeader.jsx]: href here is `/api/users/login` no need for the `localhost:3000` part

File: components/layout/components/FooterListCreator.jsx [hunk 11]
```diff
@@ -11,9 +12,7 @@ const FooterListCreator = props => {
                     const { href, text } = link;
                     return (
                         <Link href={href} key={text}>
-                            <li>
-                                <a>{text}</a>
-                            </li>
+                            <a>{text}</a>
```
Comment (Way2nnadi) [inline components/layout/components/FooterListCreator.jsx:14]: do not abuse anchor tag for css effect

Comment (justiceotuya) [reply components/layout/components/FooterListCreator.jsx:14]: How do i do it, That is Next Js way of rendering Link attribute

Comment (justiceotuya) [reply components/layout/components/FooterListCreator.jsx:14]: https://github.com/zeit/next.js/issues/5533

File: components/timeLine/selectors.js [hunk 12]
```diff
@@ -0,0 +1,7 @@
+/* eslint-disable import/prefer-default-export */
```
Comment (Way2nnadi) [inline components/timeLine/selectors.js]: add this disable to the `.eslintrc`

**Review by Way2nnadi — APPROVED** [review]
LGTM

**Pull request merged**